ambry-clustermap/src/test/java/com.github.ambry.clustermap/MockDataNodeId.java

```diff
@@ -56,6 +51,32 @@ private void populatePorts(ArrayList<Port> ports) {
     }
   }
 
+  /**
+   * Set the datacenters to which ssl is enabled. If the datacenter on which this datanode resides is
+   * part of the datacenters to which ssl is enabled, then this datanode will always return an SSL port in
```

Comment:
is there a way (or necessity) to ensure the other way round? i.e. if a node is in a DC that is ssl enabled for everyone else, then it should get an SSL port for any node that it tries to connect to.

Putting it another way, is there a way to customize this clustermap per DC? The old `getPortToConnectTo(List<String>)` provided that flexibility.